homeassistant/components/sia/alarm_control_panel.py

```diff
@@ -0,0 +1,210 @@
+                STATE_ALARM_ARMED_CUSTOM_BYPASS,
+                STATE_ALARM_ARMED_NIGHT,
+                STATE_ALARM_DISARMED,
+                STATE_ALARM_TRIGGERED,
+                STATE_UNKNOWN,
+            ]
+        ):
+            self.state = state.state
+        else:
+            self.state = None
+        await self._async_track_unavailable()
+        async_dispatcher_connect(
+            self.hass, DATA_UPDATED, self._schedule_immediate_update
+        )
+
+    @callback
+    def _schedule_immediate_update(self):
+        self.async_schedule_update_ha_state(True)
+
+    @property
+    def name(self) -> str:
+        """Get Name."""
+        return self._name
+
+    @property
+    def state(self) -> str:
+        """Get state."""
+        return self._state
+
+    @property
+    def unique_id(self) -> str:
+        """Get unique_id."""
+        return self._unique_id
+
+    @property
+    def available(self) -> bool:
+        """Get availability."""
+        return self._is_available
+
+    @property
+    def should_poll(self) -> bool:
+        """Return True if entity has to be polled for state.
+
+        False if entity pushes its state to HA.
+        """
+        return False
+
+    @property
+    def device_state_attributes(self) -> dict:
+        """Return device attributes."""
+        return self._attr
+
+    def update_attribute(self, attr: dict):
+        """Update attributes."""
+        self._attr.update(attr)
+
+    @state.setter
+    def state(self, state: str):
+        """Set state."""
+        temp = self._old_state if state == PREVIOUS_STATE else state
+        self._old_state = self._state
+        self._state = temp
+        self.async_schedule_update_ha_state()
+
+    async def assume_available(self):
+        """Reset unavalability tracker."""
+        await self._async_track_unavailable()
+
+    @callback
+    async def _async_track_unavailable(self) -> bool:
+        """Reset unavailability."""
+        if self._remove_unavailability_tracker:
+            self._remove_unavailability_tracker()
+        if self.hass:
+            self._remove_unavailability_tracker = async_track_point_in_utc_time(
+                self.hass,
+                self._async_set_unavailable,
+                utcnow() + self._ping_interval + PING_INTERVAL_MARGIN,
+            )
+        if not self._is_available:
+            self._is_available = True
+            return True
+        return False
+
+    @callback
+    def _async_set_unavailable(self, _):
+        """Set availability."""
+        self._remove_unavailability_tracker = None
+        self._is_available = False
+        _LOGGER.debug(
+            "Setting entity: %s unavailable, last heartbeat was: %s",
+            self.entity_id,
+            self._attr["last_heartbeat"],
+        )
+        self.async_schedule_update_ha_state()
+
+    @property
+    def supported_features(self) -> int:
+        """Return the list of supported features."""
+        return None
```

Comment:
Is this a passive device? You can't arm it from HA?

Reply:
Indeed, only get's updated of state changes, no way to change the state of the alarm.